src/main/java/com/github/charithe/kafka/KafkaJunitRule.java

```diff
@@ -84,8 +90,12 @@ public KafkaJunitRule(final int kafkaPort) {
 
     @Override
     protected void before() throws Throwable {
-        zookeeper = new TestingServer(true);
-        zookeeperPort = zookeeper.getPort();
+        if (zookeeperPort == -1) {
```

Comment:
In the context of this code, it is effectively allocating a random port anyway. Behind the scenes, when you instantiate a `TestingServer` without specifying the port, it uses the same code that I am using to allocate a random port to Kafka.

Anyway, you haven't used the constant in this line at all -- which was the reason for my comment in the first place.  :-)  

I think that defining two names for the same constant value is unnecessary here. Please reuse the `ALLOCATE_RANDOM_PORT` constant instead. I don't think it will be too difficult for anyone reading the code to figure out what it means in this context.